### PR TITLE
Add STATE_OPEN, STATE_CLOSED and STATE_SETTLED in Channel()

### PR DESCRIPTION
**Pull request by hrishikeshio**
Solves #494

### CODE REVIEW

**Comment by LefterisJP** [comment]
Thanks you but perhaps you can wait with this PR and rebase it on top of master after #413 is done? This will add a lot more places where the change needs to happen.

**Comment by hackaugusto** [comment]
Hi @hrishikeshio ,

#413 is merged, if you want to update your PR we can review it. Thanks again for the contribution.

**Comment by hackaugusto** [comment]
Kick note, open was renamed to opened to keep the state names consistent with settled and closed, please keep it consistent.

**Comment by hrishikeshio** [comment]
@hackaugusto Thanks. I will take note.

**Comment by hrishikeshio** [comment]
I have made the changes. Can someone please review? @hackaugusto

File: docs/api.rst [hunk 1]
```diff
@@ -49,11 +49,11 @@ A channel object consists of a
 
 - ``balance`` should be an integer of the amount of the ``token_address`` token we have available for transferring.
 
-- ``state`` should be the current state of the channel represented by a string.
+- ``state`` should be the current state of the channel represented by a constant.
   Possible value are:
-  - ``'open'``: The channel is open and tokens are tradeable
-  - ``'closed'``: The channel has been closed by a participant
-  - ``'settled'``: The channel has been closed by a participant and also settled.
+  - ``'channel.STATE_OPENED'``: The channel is open and tokens are tradeable
```
Comment (LefterisJP) [inline docs/api.rst]: Please do not change this here. The external user can't know of Python classes, he only needed to provide the constant strings as it was before. But make sure `open` -> `opened`

File: raiden/channel.py [hunk 2]
```diff
@@ -553,15 +553,18 @@ def __init__(
 
         self.received_transfers = []
         self.sent_transfers = []  #: transfers that were sent, required for settling
+        self.STATE_OPENED = 'opened'
```
Comment (LefterisJP) [inline raiden/channel.py]: I would recommend having those as static class attributes and not as members of the class instance.

To understand what I mean read [here](http://stackoverflow.com/questions/68645/static-class-variables-in-python).

No need to initialize every new channel instance with these 3 constants. Just move them to the class body and as such they should be accessible as: `Channel.STATE_OPENED` e.t.c.

File: docs/api.rst (hunk 1 above)
Comment (hrishikeshio) [inline docs/api.rst]: Ok. Will undo.

**Comment by LefterisJP** [comment]
@hrishikeshio With a quick search I can find a lot places in your branch that are still as constant strings:

```
raiden/transfer/mediated_transfer/transition.py:15:27:    if new_route.state != 'opened':
raiden/transfer/mediated_transfer/transition.py:18:29:    elif new_route.state == 'opened':
raiden/transfer/mediated_transfer/mediator.py:116:61:    payer_channel_open = transfer_pair.payer_route.state == 'opened'
raiden/transfer/mediated_transfer/mediator.py:483:56:        payee_channel_open = pair.payee_route.state == 'opened'
raiden/transfer/mediated_transfer/mediator.py:543:56:        payer_channel_open = pair.payer_route.state == 'opened'
raiden/transfer/mediated_transfer/target.py:49:40:    channel_open = from_route.state == 'opened'
raiden/transfer/state.py:33:9:        'opened',
raiden/tests/unit/transfer/mediated_transfer/factories.py:54:13:    state = 'opened'
raiden/tests/unit/transfer/mediated_transfer/test_mediator.py:199:49:        assert unpaid_pair.payer_route.state == 'opened'
raiden/tests/unit/transfer/mediated_transfer/test_mediator.py:226:47:        assert paid_pair.payer_route.state == 'opened'
raiden/tests/unit/transfer/mediated_transfer/test_mediator.py:278:43:    assert paid_pair.payer_route.state == 'opened'
raiden/tests/api/test_api.py:39:18:        'state': 'opened'
raiden/tests/api/test_api.py:87:34:    expected_response['state'] = 'opened'
raiden/tests/api/test_api.py:113:34:    expected_response['state'] = 'opened'
raiden/tests/api/test_api.py:136:18:        'state': 'opened',
raiden/tests/api/test_api.py:155:18:        'state': 'opened',
raiden/tests/api/test_api.py:186:34:    expected_response['state'] = 'opened'
raiden/channel.py:492:20:    STATE_OPENED = 'opened'
raiden/api/v1/encoding.py:151:62:    state = fields.String(validate=validate.OneOf(['closed', 'opened', 'settled']))
raiden/api/v1/encoding.py:167:44:        validate=validate.OneOf(['closed', 'opened', 'settled'])
```

**Comment by hrishikeshio** [comment]
@LefterisJP Thanks I wasn't sure if it should be done in tests. Made the changes.
However it seems importing Channel in /raiden/raiden/transfer/state.py is causing circular dependency issue. How do you suggest going about it? 
https://travis-ci.org/hrishikeshio/raiden/jobs/222287024

**Comment by hackaugusto** [comment]
@hrishikeshio define the constants in the [state.py](https://github.com/raiden-network/raiden/blob/master/raiden/transfer/state.py) module at the top-level, variable names all-caps and prefixed by the `CHANNEL_`, then import the constants from there.

At some point we might add a `Channel` object to represent a channel state and then add these constants as class attributes, but since we don't have it now this approach seems the more reasonable.

**Comment by LefterisJP** [comment]
@hrishikeshio I agree with @hackaugusto, please follow his suggestion.

**Comment by hrishikeshio** [comment]
Updated. Please review. @hackaugusto @LefterisJP

**Comment by LefterisJP** [comment]
@hrishikeshio Looks fine to me now but requires a rebase.

**Comment by hrishikeshio** [comment]
@LefterisJP Rebased and ready to merge.

**Comment by LefterisJP** [comment]
Thank you

**Pull request merged**